### DISCUSSION TITLE
Modularity, Features and Existing Packages

### DISCUSSION THREAD

**Discussion by bolducke** [orchidsoftware/platform#1211] [General]
First, I have to said that your package is very promising and a big step in the right direction. You did truly a great work there by using Screen. Thank you for your commitment

**Is your feature request related to a problem? Please describe.**
I'm having a problem with the fact that we have to use your customs features like users, permissions and attachments from your package. I would like to use spatie/laravel-medialibrary and spatie/laravel-permissions because they fit more my need in some context. By example, I would like to have Blade directives and wildcard permissions and conversion for Image processing and all.

In summary, your package is well done, but it's forcing the developer to use your own features. I don't think it's a good idea to add of the features from spatie/laravel-medialibrary to your package. It will make this package harder to maintain. Instead, by using other package, you could have the same feature with less code to write.

**Describe the solution you'd like**
I would like Orchid to be a little bit more modular. Sometimes, I don't need to use all the features from this package and I would like to use packages from other people.

My point is that it's can be tedious if I don't want to use your tools. I would like to have the choice.

**Describe alternatives you've considered**
For now, I'm not using some features from Orchid like attachment and the associated fields because I need a more complex and fully-featured way to achieve similar results.

**Comment by tabuna** [comment]
Hey, @LodrikMtl Sorry for the long silence. At the moment, throwing out `roles` or `attachment` will be a very strong blow.
This is possible in the future, but not right now. You say that you are not comfortable using them, but you can replace them yourself by installing the packages that suit you. What can I do to make it easier for you?
